src/MongoFramework/Infrastructure/Mapping/IEntityIndex.cs

```diff
@@ -9,5 +9,6 @@ public interface IEntityIndex
 		IndexSortOrder SortOrder { get; }
 		int IndexPriority { get; }
 		IndexType IndexType { get; }
+		bool IndexTenant { get; set; }
```

Comment:
Same as my other comment - not sure I like the name. Perhaps `IsTenantExclusive` or something like it?